src/Helper/Process/ProcessRunner.php

```diff
@@ -226,6 +251,13 @@ protected function createVendorBinProcess(array $command): Process {
       throw new RuntimeException(sprintf('Could not find vendor binary: %s.', $command[0]));
     }
 
+    if ($this->isEnvVarsSet === TRUE) {
+      $command = array_merge([
+        $this->envVars,
+        "php",
+      ], $command);
+    }
+
     return new Process($command);
```

Comment:
The Symfony Process component actually has special handling for environment variables. See [Setting Environment Variables for Processes](https://symfony.com/doc/current/components/process.html#setting-environment-variables-for-processes). It might look something like this, for example:

```suggestion
    return new Process($command, null, $this->envVars);
```

Of course, that assumes that `::envVars` is an array with an initial value of `[]`. Then you don't have to have `::isEnvVarsSet` at all. And you can replace `::setEnvVars()` with something a little resilient/type-safe like this, for example:

```
public function setEnv(string $name, string $value): void {
  $this->envVars[] = [$name => $value];
}
```